notebooks/opensense_data_downloader_and_transformer.py

```diff
@@ -246,30 +251,112 @@ def transform_andersson_2022_OpenMRG(fn, path_to_extract_to):
+            
+        # allocate dataset, we will read to this iteratively later
+        ds_cml2chl = xr.Dataset(
+            data_vars= dict(
+                rsl=(['sublink_id', 'channel_id', 'time'], np.zeros([
+                    int(ds.sublink.size/2), 2, timeseries.size])*np.nan),
+                tsl=(['sublink_id', 'channel_id', 'time'], np.zeros([
+                    int(ds.sublink.size/2), 2, timeseries.size])*np.nan), 
+            ),
+            coords=dict(
+                sublink_id = (df_metadata.index.values % 10000)[::2], # name of the sublinks, corresponds to ds
+                channel_id = ['channel_1', 'channel_2'],
+                time = timeseries,
+        
+                length = ('sublink_id', df_metadata.Length_km[::2]),
+                site_0_lat = ('sublink_id', df_metadata.NearLatitude_DecDeg[::2]),
+                site_0_lon = ('sublink_id', df_metadata.NearLongitude_DecDeg[::2]),
+                site_1_lat = ('sublink_id', df_metadata.FarLatitude_DecDeg[::2]),
+                site_1_lon = ('sublink_id', df_metadata.FarLongitude_DecDeg[::2]),
+                
+                frequency = (('sublink_id', 'channel_id'), df_metadata['Frequency_GHz'].values.reshape(-1, 2)),
+                polarization = (('sublink_id', 'channel_id'), df_metadata['Frequency_GHz'].values.reshape(-1, 2)),
+                
+            ),
         )
+                    
         
-    ds.attrs['comment'] += '\nMetadata added with preliminary code from opensense_data_downloader.py'
-    
-    # Change "sublink" to "sublink_id"
-    ds = ds.rename({"sublink": "sublink_id"})
-    
-    # add standard attributes
-    ds = add_cml_attributes(ds)
+        # populate dataset link by link
+        for cml in ds.sublink:
+            # metadata for inserting to ds_cml2chl
+            sublink_name = (cml.values + 1) // 2  # logic for getting the name (also index) of the link
+            sublink_channel = ((cml.values + 1) % 2) # channel index of link
+            sublink_channel = xr.where(sublink_channel == 0, 'channel_1', 'channel_2')
+
+            
+            # Below: Transform to dataframe (much faster resampling)
+            # link: https://stackoverflow.com/questions/64282393/how-can-i-speed-up-xarray-resample-much-slower-than-pandas-resample
+            
+            # 
+            if resample_1min:
+                ds_rsl = ds.sel(sublink = cml, time = slice(time_start, time_end)).rsl.to_dataframe(
+                    ).resample("1T").mean().to_xarray()
+                ds_tsl = ds.sel(sublink = cml, time = slice(time_start, time_end)).tsl.to_dataframe(
+                    ).resample("1T").mean().to_xarray()
+                
+             # resample neares probably not needed for this occation, but it ensures 
+             # that the CMLs has the same timesteps between each other and the 
+             # timeseries variable (needed in the raw file dataset)
+            else:
```

Comment:
I would not do any resampling of the raw data if the flag for resampling is not set or set to False. That is counterintuitive. And we want to keep the real raw data when doing the transformation.

I have not checked in detail, but if I recall correctly the OpenMRG CML data are all equally spaced at full 10-second time steps. Some allignment was already done when preparing the data from the real raw data files that one gets from the DAQ system, which will always have some ms or sec offsets from defined sampling intervals.

Thinking about this once more, I would now say that all sublinks in the downloadable NetCDF already are, with 100% certainty, sharing the same time index, because this is how the NetCDF is built. Hence, there is no need to resample or reindex before combining the data of the sublinks here.

Conclusion: The default behavior should be to not do resampling.